src/features/authorSlice.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+
+type State = { author: User | null };
+
+const initialState: State = { author: null };
+
+const authorSlice = createSlice({
+  name: 'author',
+  initialState,
+  reducers: {
+    setAuthor: (state, action:PayloadAction<User>) => {
+      state.author = action.payload;
```

Comment:
```suggestion
    setAuthor: (state, action:PayloadAction<User>) => {
      return {
        ...state,
        author: action.payload,
      }
```
You can avoid it like this
https://redux-toolkit.js.org/usage/immer-reducers#reducers-and-immutable-updates

Reply:
Thank You,